### PR TITLE
Add snake demo

### PR DESCRIPTION
**Pull request by shika-blyat**
I just want to say i know loading the font at every draw call when the game end is  bad, but the Task system is quite destabilizing. I spent something like 3 hours trying to load the font on the `load` function, but it seems like i missed something in the doc. 
(If you want to explain me how i am supposed to do it, i can edit my code before you merge this PR).

### CODE REVIEW

File: examples/snake.rs [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+        let mut snake = Snake::new();
+        snake.create_snake();
+        Task::succeed(|| SnakeGame {
+            snake,
+            score: 0,
+            apple: Apple::new(),
+            ticks: 0,
+            speed: 20,
+            last_key: None,
+        })
```
Comment (hecrj) [inline examples/snake.rs]: We can load the `Font` here and add it to `SnakeGame`, like this:

```rust
Font::load_from_bytes(include_bytes!(
    "../resources/font/Inconsolata-Regular.ttf"
))
.map(|font| SnakeGame {
    font,
    snake,
    score: 0,
    apple: Apple::new(),
    ticks: 0,
    speed: 20,
    last_key: None,
})
```

File: examples/snake.rs [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+            direction: None,
+        }
+    }
+    fn create_snake(&mut self) {
+        for i in 1..6 {
+            let x = (30.0 * i as f32) + 30.0;
+            let y = 90.0;
+            self.square_pos.push(Position(x, y));
+        }
+    }
```
Comment (hecrj) [inline examples/snake.rs]: We could move this logic to `new`. This way we avoid the scenario of "empty snakes".

File: examples/snake.rs [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+            });
+            font.add(Text {
+                content: format!("Your score is: {}", self.score).as_str(),
+                position: Point::new(270.0, 200.0),
+                size: 50.0,
+                color: Color::WHITE,
+                ..Text::default()
+            });
+            font.draw(&mut frame.as_target());
+        }
```
Comment (hecrj) [inline examples/snake.rs]: We could use [`coffee::ui`](https://docs.rs/coffee/0.3.2/coffee/ui/index.html) to draw the game over screen and maybe the current score during gameplay.

File: examples/snake.rs [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+        maximized: false,
+        fullscreen: false,
+    })
+    .expect("An error occured while starting the game");
+}
+
+fn new_random_pos() -> (f32, f32) {
+    let mut rng = rand::thread_rng();
+    let x = (0..900).step_by(30).choose(&mut rng).unwrap() as f32;
+    let y = (0..600).step_by(30).choose(&mut rng).unwrap() as f32;
```
Comment (hecrj) [inline examples/snake.rs]: May be a good idea to define a couple of constants for the window dimensions.

Comment (shika-blyat) [reply examples/snake.rs]: I'll reply only one time for all the change request, but they're all good ideas. I'll try to make these change tonight or tommorow. There's just a problem about Point. I tried to use at first but I had errors about alias used as constructor in some case. So idk if I should use Point where I can use it and use Position the rest of the time or just keeper with Position everywhere.

**Comment by shika-blyat** [comment]
I wanted to work on it, but i realize since i restarted school i don't have much time. I think i wouldn't have the time to work on it before 5/6 weeks. If anyone want to finish my work, he can. Also, if you want to merge it like this, i still can work on improving the code later :)

**Comment by hecrj** [comment]
@shika-blyat I understand!

Let's merge it! I may try to improve it myself if I find some time (very busy with [`iced`]() lately :sweat_smile:) Thank you again! <3

**Pull request merged**